x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+        expect(CredentialsLogic.values.nameInputBlurred).toEqual(true);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setNameInputBlurred(true);
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        nameInputBlurred: expect.any(Boolean),
+      });
+    });
+  });
+
+  describe('setTokenReadWrite', () => {
+    describe('activeApiToken', () => {
+      // TODO should probably have a more specific behavior
+      it('sets an arbitrary key and boolean value on the activeApiToken', () => {
+        mount({
+          activeApiToken: {
+            ...newToken,
+            foo: 'bar',
+          },
+        });
+        CredentialsLogic.actions.setTokenReadWrite({ name: 'foo', checked: true });
+        expect(CredentialsLogic.values.activeApiToken).toEqual({
+          ...newToken,
+          foo: true,
+        });
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setTokenReadWrite({ name: 'foo', checked: true });
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+      });
+    });
+  });
+
+  describe('setTokenName', () => {
+    describe('activeApiToken', () => {
+      it('update the name property on the activeApiToken, formatted correctly', () => {
+        mount({
+          activeApiToken: {
+            ...newToken,
+            name: 'bar',
+          },
+        });
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.activeApiToken).toEqual({
+          ...newToken,
+          name: 'new-name',
+        });
+      });
+    });
+
+    describe('activeApiTokenRawName', () => {
+      it('updates the raw name, with no formatting applied', () => {
+        mount();
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual('New Name');
+      });
+    });
+
+    describe('keyHelpText', () => {
+      it('should update the keyHelpText value to reflect the new name', () => {
+        mount();
+        CredentialsLogic.actions.setTokenName('New Name');
+        expect(CredentialsLogic.values.keyHelpText).toEqual('Your key will be named: new-name');
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setTokenName('New Name');
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+        activeApiTokenRawName: expect.any(String),
+        keyHelpText: expect.any(String),
+      });
+    });
+  });
+
+  describe('setTokenType', () => {
+    describe('activeApiToken.access_all_engines', () => {
+      describe('when value is ADMIN', () => {
+        it('updates access_all_engines to false', () => {
```

Comment:
This reads weird to me - shouldn't admin keys have access to all engines? (not saying this test is wrong btw, just that the root behavior seems odd)

Reply:
edit: i forgot admin keys don't apply to engines, disregard me. deleting above comment :doh: